script/make_release.py

```diff
@@ -0,0 +1,262 @@
+import subprocess
+import threading
+import os
+import shutil
+import uuid
+import stat
+import concurrent.futures
+import argparse
+import logging
+import platform
+
+DEFAULT_ARCHS = [
+    'x86-64-vnni512',
+    'x86-64-vnni256',
+    'x86-64-avx512',
+    'x86-64-bmi2',
+    'x86-64-avx2',
+    'x86-64-sse41-popcnt',
+    'x86-64-modern',
+    'x86-64-ssse3',
+    'x86-64-sse3-popcnt',
+    'x86-64',
+    'x86-32-sse41-popcnt',
+    'x86-32-sse2',
+    'x86-32',
+    'general-64',
+    'general-32'
+]
+
+parser = argparse.ArgumentParser(description='Stockfish release maker', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('--sde', default=None, type=str,
+                    help='Path to Intel SDE. If not specified it will not be used. If specified it will be used as a fallback for architectures that fail to compile.')
+parser.add_argument('--concurrency', default=1, type=int,
+                    help='The maximum number of jobs to run concurrently.')
+parser.add_argument('--repository', default='https://github.com/official-stockfish/Stockfish.git', type=str,
+                    help='The path to the repository to clone.')
+parser.add_argument('--clone-dir', default='repo', type=str,
+                    help='The directory to clone the repository to.')
+parser.add_argument('--out-dir', default='bin', type=str,
+                    help='The directory that shall contain the resulting binaries.')
+parser.add_argument('--build-type', default='profile-build', type=str,
+                    help='The build type to use in make.')
+parser.add_argument('--prefix', default='', type=str,
+                    help='The string put in the binary name after stockfish- but before -arch. For example "win" or "linux". Will be ommited if empty.')
+parser.add_argument('--verbose', default=False, action='store_true',
+                    help='Keep output of running programs.')
+parser.add_argument('--keep-logs', default=False, action='store_true',
+                    help='Whether to keep logs in the output directory. Incompatible with --verbose.')
+parser.add_argument('archs', nargs='*', default=DEFAULT_ARCHS, type=str,
+                    help='A list of architectures to compile. If not provided then most sensible archs will be built.')
+
+args = parser.parse_args()
+
+if args.verbose and args.keep_logs:
+    parser.error('--verbose and --keep-logs are currently not compatible.')
+
+SDE_PATH = args.sde
+NUM_THREADS = args.concurrency
+REPO_DIR = args.clone_dir
+OUT_DIR = args.out_dir
+REPO = args.repository
+BUILD_TYPE = args.build_type
+ARCHS = args.archs
+STDOUT = None if args.verbose else subprocess.DEVNULL
+STDERR = subprocess.STDOUT
+PREFIX = args.prefix
+IS_WINDOWS = 'windows' in platform.system().lower()
+EXE_SUFFIX = '.exe' if IS_WINDOWS else ''
+
+def log(s):
```

Comment:
I will prefer to use directly `print()`.

For advanced use there is `logging`, see https://docs.python.org/3.6/howto/logging.html